### PR TITLE
[WIP] fix(bootstrap): call the CD only once

### LINKED ISSUE #3701: ng-if conditions are called twice, should also short-circuit?
Not sure if this actually a bug or not but given, http://plnkr.co/edit/exgKLQWZVQjyg5vgnqI8 the conditionals in `ng-if` appear to be called twice each. 

Additionally in JS `||` and `&&` operators both short-circuit, in `ng-if` both sides are evaluated regardless. Again I'm not entirely sure if this is the expected behaviour or not, was hoping someone could shed some light.

### PR DESCRIPTION
**Pull request by vicb**
fixes #3701

### CODE REVIEW

File: modules/angular2/src/core/life_cycle/life_cycle.ts [hunk 1]
```diff
@@ -47,29 +45,26 @@ export class LifeCycle {
...
-  registerWith(zone: NgZone, changeDetector: ChangeDetector = null) {
+  registerWith(zone: NgZone, changeDetector: ChangeDetector = null): void {
     if (isPresent(changeDetector)) {
       this._changeDetector = changeDetector;
     }
-    zone.overrideOnTurnDone(() => this.tick());
+
+    // `tick()` must be called automatically starting from the second turn because it is called
+    // explicitely by the bootstrap process for the first turn.
+    zone.overrideOnTurnDone(() => { zone.overrideOnTurnDone(() => this.tick()); });
```
Comment (vicb) [inline modules/angular2/src/core/life_cycle/life_cycle.ts]: This is the important change where we delay automatic calls to `tick()`.

**Comment by vicb** [comment]
Seems like this PR needs more work for Dart also some e2e tests are failing...

**Comment by tbosch** [comment]
@vicb What is the status here?

**Comment by vicb** [comment]
I've been working on the API doc which is higher priority, I'll resume when I'm done with the docs.

**Comment by vicb** [comment]
closing - not working on this atm

**Pull request closed without merging**